fastestimator/summary/system.py

```diff
@@ -74,17 +75,43 @@ def __init__(self,
 
         self.network = network
         self.mode = mode
-        self.global_step = None
         self.num_devices = num_devices
         self.log_steps = log_steps
         self.total_epochs = total_epochs
-        self.epoch_idx = None
         self.batch_idx = None
         self.max_train_steps_per_epoch = max_train_steps_per_epoch
         self.max_eval_steps_per_epoch = max_eval_steps_per_epoch
         self.stop_training = False
         self.summary = Summary(None)
         self.experiment_time = ""
+        self._initialize_state()
+
+    def _initialize_state(self) -> None:
+        """Initialize the training state.
+        """
+        self.global_step = None
+        self.epoch_idx = 0
+
+    def load_state(self, json_path):
+        """Load training state.
+
+        Args:
+            json_path: The json file path to load from.
+        """
+        with open(json_path, 'r') as fp:
+            state = json.load(fp)
+        self.epoch_idx = state["epoch_idx"]
+        self.global_step = state["global_step"]
+
+    def save_state(self, json_path):
+        """Load training state.
+
+        Args:
+            json_path: The json file path to save to.
+        """
+        state = {"epoch_idx": self.epoch_idx, "global_step": self.global_step}
+        with open(json_path, 'w') as fp:
+            json.dump(state, fp, indent=4)
```

Comment:
why not dump/load the entire state dictionary?

Reply:
you mean system?  we don't want to overwrite some other system args like `num_devices`, `log_steps` and so on.

Reply:
we should also keep "stop_training", "summary" and "experiment_time"

Reply:
added them in # TODO :)

Reply:
lol